### ISSUE TITLE
example notebooks

### ISSUE THREAD

**Issue by frankherfert** [dreamquark-ai/tabnet#161]
I have gone through the examples and made a few changes, which I would like to submit a pull request for:
- in one of the datasets, the first row was accidentally used as a header
- in some cases, the label encoding was overwriting continuous variables with few unique values
- I have added a few helper functions for detecting categorical variables, getting their column index and getting embedding dimensions
- short benchmark on different embedding sizes for Adult Census Income dataset
- added comparison for LightGBM next to XGBoost
- added parts to compare feature importance from TabNet, XGBoost and LightGBM
- added links to open notebooks in Google Colab in the readme

**Comment by Optimox** [comment]
Hello @frankherfert,

Thanks for opening this issue! Indeed the notebooks examples can be improved. Feel free to create a PR with your changes so that we can discuss them in detail.

Please also have a look at this PR as it tackles the same examples: #158

**Comment by frankherfert** [comment]
Thanks for the quick reply.
There is some overlap with #158. I didn't see this one before I started on the notebooks.

I will submit a PR and mention them in the other issue, so we can discuss which parts to use.
